service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/GetAzureDiskAttachedVmStepTest.java

```diff
@@ -0,0 +1,123 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@Tag("unit")
+@ExtendWith(MockitoExtension.class)
+class GetAzureDiskAttachedVmStepTest {
+  private static final String DISK_NAME = "dataDisk";
+  private static final String AZURE_SUBSCRIPTION_ID = "subscriptionId";
+  private static final String AZURE_RESOURCE_GROUP_ID = "resourceGroupId";
+
+  @Mock private AzureConfiguration azureConfigurationMock;
+  @Mock private CrlService crlServiceMock;
+  @Mock private ControlledAzureDiskResource controlledAzureDiskResourceMock;
+  @Mock private FlightContext flightContextMock;
+  @Mock private AzureCloudContext azureCloudContextMock;
+  @Mock private FlightMap flightWorkingMapMock;
+  @Mock private ComputeManager computeManagerMock;
+  @Mock private Disks disksMock;
+  @Mock private Disk diskMock;
+
+  private GetAzureDiskAttachedVmStep getAzureDiskAttachedVmStep;
+
+  @BeforeEach
+  void setup() {
+    getAzureDiskAttachedVmStep =
+        new GetAzureDiskAttachedVmStep(
+            azureConfigurationMock, crlServiceMock, controlledAzureDiskResourceMock);
+  }
+
+  @Test
+  void doStep_VmIdentifierSaved() throws InterruptedException {
+    final String vmId = "vmId";
+    setupCommonMocks();
+    when(disksMock.getById(anyString())).thenReturn(diskMock);
+    when(diskMock.isAttachedToVirtualMachine()).thenReturn(true);
+    when(diskMock.virtualMachineId()).thenReturn(vmId);
+    StepResult stepResult = getAzureDiskAttachedVmStep.doStep(flightContextMock);
+
+    assertThat(stepResult, equalTo(StepResult.getStepResultSuccess()));
+    verify(flightWorkingMapMock, times(1))
+        .put(DeleteAzureDiskFlightUtils.DISK_ATTACHED_VM_ID_KEY, vmId);
+  }
+
+  @Test
+  void doStep_DiskIsNotAttached() throws InterruptedException {
+    setupCommonMocks();
+    when(disksMock.getById(anyString())).thenReturn(diskMock);
+    StepResult stepResult = getAzureDiskAttachedVmStep.doStep(flightContextMock);
+
+    assertThat(stepResult, equalTo(StepResult.getStepResultSuccess()));
+    verify(flightWorkingMapMock, never()).put(any(), any());
+  }
+
+  @Test
+  void doStep_DiskIsNotFound() throws InterruptedException {
+    setupCommonMocks();
+    ManagementException resourceNotFoundExceptionMock =
+        setupManagementExceptionMock("ResourceNotFound");
+    when(disksMock.getById(anyString())).thenThrow(resourceNotFoundExceptionMock);
+
+    StepResult stepResult = getAzureDiskAttachedVmStep.doStep(flightContextMock);
+    assertThat(stepResult, equalTo(StepResult.getStepResultSuccess()));
+    verify(flightWorkingMapMock, never()).put(any(), any());
+  }
+
+  @Test
+  void undoStep_Success() throws InterruptedException {
+    // always return success
+    assertThat(
+        getAzureDiskAttachedVmStep.undoStep(flightContextMock),
+        equalTo(StepResult.getStepResultSuccess()));
+  }
+
+  private void setupCommonMocks() {
+    when(controlledAzureDiskResourceMock.getDiskName()).thenReturn(DISK_NAME);
```

Comment:
If this is happening in every test, why not just include it in the setup method?

Reply:
@blakery not sure what exactly do you mean here. It looks like I already moved all common code into `setupCommonMocks`. No?